### PR TITLE
Make ks-apiserver be easier to run locally with kube config

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
As I mentioned in the title, users can run ks-apiserver locally.

**Which issue(s) this PR fixes**:
None.

**Special notes for reviewers**:
None.

**Additional documentation, usage docs, etc.**:
None.

### CODE REVIEW

**Comment by wansir** [comment]
`clientcmd.DeferredLoadingClientConfig` may be a good choice.

https://github.com/kubesphere/kubesphere/blob/master/vendor/sigs.k8s.io/controller-runtime/pkg/client/config/config.go#L120-L137

File: pkg/simple/client/k8s/options.go [hunk 1]
```diff
@@ -44,12 +47,18 @@ type KubernetesOptions struct {
...
-		Burst:      1e6,
+func NewKubernetesOptions() (option *KubernetesOptions) {
+	option = &KubernetesOptions{
+		QPS:   1e6,
+		Burst: 1e6,
+	}
+
+	// make it be easier for those who wants to run api-server locally
+	userHomeConfig := path.Join(homedir.HomeDir(), ".kube/config")
+	if _, err := os.Stat(userHomeConfig); !os.IsNotExist(err) {
```
Comment (yuswift) [inline pkg/simple/client/k8s/options.go:58]: I'd prefer to judge if err==nil. Why do you only handle the `NotExist` err?

Comment (LinuxSuRen) [reply pkg/simple/client/k8s/options.go]: Excellent suggestion!

**Comment by LinuxSuRen** [comment]
> `clientcmd.DeferredLoadingClientConfig` may be a good choice.

Thanks for your suggestion. Here we only need the KubeConfig file path. I learned from your suggested code lines.

File: pkg/simple/client/k8s/options.go [hunk 2]
```diff
@@ -44,12 +48,26 @@ type KubernetesOptions struct {
...
+		if u, err := user.Current(); err == nil {
+			homePath = u.HomeDir
+		}
+	}
+
+	userHomeConfig := path.Join(homePath, ".kube/config")
+	if _, err := os.Stat(userHomeConfig); err == nil {
+		option.KubeConfig = userHomeConfig
 	}
+	return
```
Comment (JohnNiang) [inline pkg/simple/client/k8s/options.go:70]: IMO, the `return` line is unnecessary for here.

Comment (LinuxSuRen) [reply pkg/simple/client/k8s/options.go:70]: image

Comment (JohnNiang) [reply pkg/simple/client/k8s/options.go:70]: I have no problem as https://github.com/golang/go/issues/12168

Comment (LinuxSuRen) [reply pkg/simple/client/k8s/options.go:70]: > I have no problem as [golang/go#12168](https://github.com/golang/go/issues/12168)

I didn't get your point. Are you trying to say that there's no error in your IDE?

Comment (JohnNiang) [reply pkg/simple/client/k8s/options.go:70]: I mean that I've seen the [declined proposal](https://github.com/golang/go/issues/12168) about return statement at the very end of a function, so I have no problem of this return statement.

**Pull request merged**